openhcl/underhill_core/src/worker.rs

```diff
@@ -1064,6 +1064,12 @@ impl LoadedVmNetworkSettings for UhVmNetworkSettings {
     async fn save(&mut self) -> Vec<ManaSavedState> {
         let mut vf_managers: Vec<(Guid, Arc<HclNetworkVFManager>)> =
             self.vf_managers.drain().collect();
+
+        // Nothing to save
+        if vf_managers.is_empty() {
```

Comment:
Instead of saving null vector, can we return `Option<Vec<ManaSavedState>>` from this function and save that directly? And, in this case, it would just return None, so that no state is saved